### PR TITLE
Update to Go 1.22 and latest Go dependencies

### PR DESCRIPTION
**Pull request by embik**
This PR does a couple of things:

1. Bump the Go version installed in CI to 1.22. Go 1.20 is [EOL](https://endoflife.date/go) and won't receive further updates.
2. Update various k8s.io dependencies. In specific, client-go was using the old v11.0.0 version, now it uses the 1.29 version of the library.
3. Update spf13/cobra from 1.8 to 1.9.
4. Update golangci-lint and fix code that newer linter versions complain about.

Basically, spring cleaning for the code base.

### CODE REVIEW

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!
